spec/loophp/collection/CollectionSpec.php

```diff
@@ -37,6 +38,13 @@ class CollectionSpec extends ObjectBehavior
 {
     private const PHP_8 = 80_000;
 
+    public function getMatchers(): array
+    {
+        return [
+            'beSameAs' => static fn (CollectionInterface $subject, CollectionInterface $expected): bool => $subject->same($expected),
+        ];
+    }
```

Comment:
adding a custom matcher based on: https://www.phpspec.net/en/stable/cookbook/matchers.html#inline-matcher

Couldn't find a better way to test the function, let me know if you can think of any

Reply:
This is a good way indeed! And I don't see any other option at the moment.